src/main/java/io/cdap/plugin/date/DateTransformConfig.java

```diff
@@ -0,0 +1,290 @@
+
+  @Name("schema")
+  @Description("Specifies the schema of the records outputted from this plugin.")
+  private String schema;
+
+  @VisibleForTesting
+  public DateTransformConfig(String sourceFields, @Nullable String sourceFormat,
+                             String targetFields, @Nullable String targetFormat,
+                             @Nullable String secondsOrMilliseconds, String schema) {
+    this.sourceFields = sourceFields;
+    this.sourceFormat = sourceFormat;
+    this.targetFields = targetFields;
+    this.targetFormat = targetFormat;
+    this.secondsOrMilliseconds = (secondsOrMilliseconds == null) ? "Milliseconds" : secondsOrMilliseconds;
+    this.schema = schema;
+  }
+
+  private DateTransformConfig(Builder builder) {
+    this.sourceFields = builder.sourceFields;
+    this.sourceFormat = builder.sourceFormat;
+    this.targetFields = builder.targetFields;
+    this.targetFormat = builder.targetFormat;
+    this.secondsOrMilliseconds = builder.secondsOrMilliseconds;
+    this.schema = builder.schema;
+  }
+
+  public String getSourceFormat() {
+    return sourceFormat;
+  }
+
+  public String getTargetFormat() {
+    return targetFormat;
+  }
+
+  public boolean isInSeconds() {
+    if (secondsOrMilliseconds == null) {
+      return false;
+    }
+    return secondsOrMilliseconds.equals("Seconds");
+  }
+
+  public String getSecondsOrMilliseconds() {
+    return secondsOrMilliseconds;
+  }
+
+  public List<String> getSourceFields() {
+    String[] sourceFields = this.sourceFields.split(",");
+    List<String> stringList = new ArrayList<>();
+    for (String field : sourceFields) {
+      stringList.add(field.trim());
+    }
+    return stringList;
+  }
+
+  public List<String> getTargetFields() {
+    String[] sourceFields = targetFields.split(",");
+    List<String> stringList = new ArrayList<>();
+    for (String field : sourceFields) {
+      stringList.add(field.trim());
+    }
+    return stringList;
+  }
+
+  public String getTargetFieldsString() {
+    return targetFields;
+  }
+
+  public String getSourceFieldsString() {
+    return sourceFields;
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    try {
+      Schema outputSchema = getSchema();
+      if (!containsMacro(PROPERTY_TARGET_FIELDS)) {
+        List<String> targetFields = this.getTargetFields();
+        for (String targetField : targetFields) {
+          if (outputSchema.getField(targetField) == null) {
+            failureCollector.addFailure(String.format("Target field '%s' is not present in output schema",
+                                                      targetField), null)
+              .withConfigProperty(PROPERTY_TARGET_FIELDS)
+              .withOutputSchemaField(targetField);
+          }
+        }
+      }
+    } catch (IllegalStateException e) {
+      failureCollector.addFailure("Output schema cannot be parsed.", null)
+        .withStacktrace(e.getStackTrace());
```

Comment:
`.withConfigProperty`

Reply:
There is not config property. It's just schema json cannot be parsed.

Reply:
we use "schema" as config property.